configuration/configuration_blackbox_test.go

```diff
@@ -31,7 +40,7 @@ func TestMain(m *testing.M) {
 
 func resetConfiguration() {
 	var err error
-	config, err = configuration.GetConfigurationData()
+	config, err = configuration.NewConfigurationData("../config.yaml")
```

Comment:
> Now, the values in (1) config.yaml ( present in repo ) executed during tests are exactly the same as the (2) defaults declared in the `configuration.go' . So, whether or not we pass the path hardly makes any difference in tests.

It will make difference if we change the defaults but won't update the config.yaml. The test will pass but it may fail in prod or make dev when you don't use any configuration file there.